pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -914,6 +925,34 @@ type PrometheusSpec struct {
 	TSDB TSDBSpec `json:"tsdb,omitempty"`
 }
 
+type WhenScaledRetentionType string
+
+var (
+	WhenScaledRetentionTypeRetain WhenScaledRetentionType = "retain"
+	WhenScaledRetentionTypeDelete WhenScaledRetentionType = "delete"
+)
+
+type ShardRetentionPolicy struct {
+	// WhenScaled defines the retention policy when the Prometheus Server is scaled-down.
+	// When set to retain, the statefulset will only stop scraping targets and will retain the data.
+	// The remaining shards' scrape configuration will be updated to scrape the targets of the scaled-down shard.
+	//
+	// For backwards compatibility, 'delete' is the default value. In a future v2 API, 'retain' will be the default value.
+	//
+	// Default: "delete"
+	// +kubebuilder:validation:Enum=retain;delete
+	// +required
+	WhenScaled string `json:"whenScaled"`
+
+	// Retention defines the retention period of the Prometheus Server.
```

Comment:
The "retention" term here is confusing with the actual Prometheus retention settings. Not sure if we can find something more explicit (`deleteAfter`, `keepFor`, `timeToLive`?).

To keep things simple, Maybe we can skip this field in this PR?

Reply:
> To keep things simple, Maybe we can skip this field in this PR?

To make it easier to review? Yeah, I don't see any problems with that :)

Reply:
Yes I was mostly concerned with the review load :)